contributions/MPI_IS_gaussian_process/CMakeLists.txt

```diff
@@ -82,34 +82,54 @@ set_property(TARGET GPGuider PROPERTY FOLDER "Contributions/")
 
 # Test for the GP
 add_executable(GaussianProcessTest ${gaussian_process_root_dir}/tests/gaussian_process/gaussian_process_test.cpp)
-target_link_libraries(GaussianProcessTest MPIIS_GP gtest)
+if (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    target_link_libraries(GaussianProcessTest MPIIS_GP GTest::GTest)
+else()
+    target_link_libraries(GaussianProcessTest MPIIS_GP gtest)
+endif()
 target_include_directories(GaussianProcessTest  PRIVATE ${gaussian_process_root_dir}/tools ${GTEST_HEADERS})
 set_property(TARGET GaussianProcessTest PROPERTY FOLDER "Unit tests/Contribution")
 add_test(NAME GaussianProcessTest COMMAND GaussianProcessTest)
 
 # Test for the math tools
 add_executable(MathToolboxTest ${gaussian_process_root_dir}/tests/gaussian_process/math_tools_test.cpp)
-target_link_libraries(MathToolboxTest MPIIS_GP_TOOLS gtest)
+if (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    target_link_libraries(MathToolboxTest MPIIS_GP_TOOLS GTest::GTest)
+else()
+    target_link_libraries(MathToolboxTest MPIIS_GP_TOOLS gtest)
+endif()

```

Comment:
Instead of all those `if/else/endif`, you may define once a variable like `gtest_link` as follow
```
if (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
    set(gtest_link "GTest::GTest")
else()
    set(gtest_link "gtest")
endif()
```

and use it like this
```
target_link_libraries(MathToolboxTest MPIIS_GP_TOOLS ${gtest_link})
```

Reply:
Sure, no problem. It's late here so I'll look into it tomorrow.

The "correct" way to do this, in my opinion would be migrating the Linux build to GTest::GTest as well since this is what find_package sets for Google Test. This would require some modifications to the way the bundled Google Test is built so maybe the variable is preferable?

Reply:
I added #839 so we can fix that separately. I'll leave it up to you whether you want to tackle that or not in this PR -- if not then @raffienficiaud's variable suggestion is fine for now.

Reply:
I opted for the variable solution for now.